File: Pattern 17/Pattern_17.java
class Solution {

    void printTriangle(int n) {
        // same as previous question
        // in each row have element that is equal to 2*i+1
        // in zero based indexing
        // haft +1 character are increase and remaing are decreasing
        for(int i =0; i<n; i++) {
            // spaces
            for(int j = n-1; j>i; j--) {
                System.out.print(" ");
            }
            char el = 'A';
            // character
            for(int j =0; j<2*i+1; j++) {
                System.out.print(el);
                if(j<i) el++;
                else el--;
            }
            System.out.println();
        }
    }
}

File: Pattern 17/Pattern_17.py
class Solution:
    def printTriangle(self, n):
        # same as previous question
        # in each row have element that is equal to 2*i+1
        # in zero based indexing
        # haft +1 character are increase and remaing are decreasing
        for i in range(0,n):
        
            # spaces
            for j in range(n-1,i,-1):
                print(end= " ")
                
            el = 65
            # character
            for j in range(0,2*i+1):
                print(chr(el),end="")
                if j < i:  
                    el += 1
                else:
                    el -= 1
            print()
 